sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -316,17 +301,85 @@ object GpuParquetScan {
    * @return the updated target batch size.
    */
   def splitTargetBatchSize(targetBatchSize: Long, useChunkedReader: Boolean): Long = {
-      if (!useChunkedReader) {
+    if (!useChunkedReader) {
       throw new SplitAndRetryOOM("GPU OutOfMemory: could not split inputs " +
-          "chunked parquet reader is configured off")
+        "chunked parquet reader is configured off")
     }
     val ret = targetBatchSize / 2
     if (targetBatchSize < minTargetBatchSizeMiB * 1024 * 1024) {
-           throw new SplitAndRetryOOM("GPU OutOfMemory: could not split input " +
-          s"target batch size to less than $minTargetBatchSizeMiB MiB")
+      throw new SplitAndRetryOOM("GPU OutOfMemory: could not split input " +
+        s"target batch size to less than $minTargetBatchSizeMiB MiB")
     }
     ret
   }
+
+  def throwIfRebaseNeededInExceptionMode(table: Table, dateRebaseMode: DateTimeRebaseMode,
+      timestampRebaseMode: DateTimeRebaseMode): Unit = {
+    (0 until table.getNumberOfColumns).foreach { i =>
+      val col = table.getColumn(i)
+      if (dateRebaseMode == DateTimeRebaseException &&
+        DateTimeRebaseUtils.isDateRebaseNeededInRead(col)) {
+        throw DataSourceUtils.newRebaseExceptionInRead("Parquet")
+      }
+      else if (timestampRebaseMode == DateTimeRebaseException &&
+        DateTimeRebaseUtils.isTimeRebaseNeededInRead(col)) {
+        throw DataSourceUtils.newRebaseExceptionInRead("Parquet")
+      }
+    }
+  }
+
+  def rebaseDateTime(table: Table, dateRebaseMode: DateTimeRebaseMode,
+      timestampRebaseMode: DateTimeRebaseMode): Table = {
+    val tableHasDate = (0 until table.getNumberOfColumns).exists { i =>
+      checkTypeRecursively(table.getColumn(i), { dtype => dtype == DType.TIMESTAMP_DAYS })
+    }
+    val tableHasTimestamp = (0 until table.getNumberOfColumns).exists { i =>
+      checkTypeRecursively(table.getColumn(i), { dtype => dtype.isTimestampType })
+    }
+
+    if ((tableHasDate && dateRebaseMode == DateTimeRebaseLegacy) ||
```

Comment:
What if I have a different config for date and for timestamp? I don't know how common it is, but we cannot support it properly with this code.

Reply:
From Spark code, I believe that we can have different configs but only when we read/write timestamp as int96. You're right, the current transform function doesn't handle different configs for date and time. I'll update it.

Reply:
Done.